util/panics/panics.go

```diff
@@ -1,69 +1,61 @@
 package panics
 
 import (
+	"fmt"
 	"github.com/kaspanet/kaspad/logs"
-	"github.com/kaspanet/kaspad/signal"
+	"os"
 	"runtime/debug"
 	"time"
 )
 
 // HandlePanic recovers panics, log them, runs an optional panicHandler,
 // and then initiates a clean shutdown.
-func HandlePanic(log logs.Logger, goroutineStackTrace []byte, panicHandler func()) {
-	if err := recover(); err != nil {
+func HandlePanic(backend *logs.Backend, goroutineStackTrace []byte) {
+	const panicSubsystem = "PANC"
+
+	log := backend.Logger(panicSubsystem)
```

Comment:
Please move the above two lines to after we've checked that there really is a panic.